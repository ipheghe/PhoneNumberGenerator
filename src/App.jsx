import React, { Component } from 'react';

import Header from './components/Header';

/**
 * Container component housing the Single-Page application
 *
 * @class Dashboard
 * @extends {React.Component<DashboardProps, DashboardState>}
 */export default class App extends Component {
  /**
   * Creates an instance of App
   *
   * @param {object} props - props
   * @memberof App
   * @returns {void}
   */
  constructor(props) {
    super(props);

    this.state = {
      welcomeText: 'Welcome',
    };
  }

  /**
   * Renders App component
   *
   * @returns {JSX} jsx
   * @memberof App
   */
  render() {
    return (
      <div>
        <Header />
      </div>
    );
  }
}
