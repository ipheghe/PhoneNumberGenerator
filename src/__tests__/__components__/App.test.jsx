// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import render from 'react-test-renderer';
import { spy } from 'sinon';

// components
import App from '../../App';

const event = {
  target: {
    name: 'chosenNumber',
    value: 3,
    selectedOptions: [
      { innerHTML: 'selected name' },
    ],
  },
  preventDefault: spy(),
};

describe('App component', () => {
  it('should match component snapshot', () => {
    const tree = render.create(<App />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const shallowWrapper = shallow(<App />);
    expect(shallowWrapper).toBeDefined();
    expect(shallowWrapper.find('div').length).toBe(1);
    expect(shallowWrapper.exists()).toBe(true);
  });

  it('should call handleChange method', () => {
    const shallowWrapper = shallow(<App />);
    const handleChangeSpy = spy(shallowWrapper.instance(), 'handleChange');

    expect(shallowWrapper.state().chosenNumber).toEqual(0);

    shallowWrapper.instance().handleChange(event);
    expect(handleChangeSpy.called).toEqual(true);
    expect(shallowWrapper.state().chosenNumber).toEqual(3);
  });

  it('should call handleButtonClick method', () => {
    const shallowWrapper = shallow(<App />);
    const handleButtonClickSpy = spy(shallowWrapper.instance(), 'handleButtonClick');

    expect(shallowWrapper.state().buttonText).toEqual('Generate');

    shallowWrapper.instance().handleButtonClick(event);
    expect(handleButtonClickSpy.called).toEqual(true);
    expect(shallowWrapper.state().buttonText).toEqual('Generating Numbers.....');
  });
});
