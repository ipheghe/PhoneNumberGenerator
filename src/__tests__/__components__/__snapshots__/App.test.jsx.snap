// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component rshould match component snapshot 1`] = `
<div>
  <div
    className="header__container"
  >
    <h3>
      Number Generator
    </h3>
    <ul
      className="menu__container"
    >
      <li
        className="menu__container__items"
      >
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="menu__container__items"
      >
        <a
          href="/operations"
          onClick={[Function]}
        >
          Operations
        </a>
      </li>
    </ul>
  </div>
</div>
`;
